### PR TITLE
LF-4216 - remove purpose check for future see LF-4246

### PR DESCRIPTION
**Pull request by Duncan-Brain**
**Description**

Removes purpose check in favour of moving forward Soil Health V1

Jira link: 

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js [hunk 1]
```diff
@@ -238,11 +239,12 @@ export const up = async function (knex) {
     table.integer('purpose_id').references('id').inTable('soil_amendment_purpose').notNullable();
     table.primary(['task_products_id', 'purpose_id']);
     table.string('other_purpose');
-    table.check(
-      '(?? IS NOT NULL AND ?? = ??) OR (?? IS NULL)',
-      ['other_purpose', 'purpose_id', otherPurpose.id, 'other_purpose'],
-      'other_purpose_id_check',
-    );
+    // LF-4246 - add back in this check
```
Comment (kathyavini) [inline packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js:242]: You want to comment it? I think there's a chance the outcome of the LF-4246 investigation will be that we don't really care to do more migration on this table. Clean delete would let us go either way. This kinda feels like it locks us into another migration 😂

Comment (Duncan-Brain) [reply packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js:242]: Yeah I think it is helpful context. 🤷

**Review by SayakaOno — APPROVED** [review]
I thought we were going to delete existing `other_purpose` where the purpose isn't "other" in this PR!
Looks good!

File: packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js (hunk 1 above)
Comment (kathyavini) [reply packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js:242]: Ok. My preference not that strong 🙂

**Review by kathyavini — APPROVED** [review]

**Comment by Duncan-Brain** [comment]
Ah sorry @SayakaOno we miscommunicated! Added a comment for David to think on here: https://lite-farm.atlassian.net/browse/LF-4246

**Pull request merged**